addons/test/testutil/test_helpers.go

```diff
@@ -266,19 +266,21 @@ func getResource(decoder *yamlutil.YAMLOrJSONDecoder, mapper meta.RESTMapper, dy
 
 // GetExternalCRDPaths gets paths for external CRDs by introspecting versions of the go dependencies
 func GetExternalCRDPaths(externalDeps map[string][]string) ([]string, error) {
-	var crdPaths []string
-	gopath, err := exec.Command("go", "env", "GOPATH").Output()
-	if err != nil {
-		return crdPaths, err
+	packageConfig := &packages.Config{
+		Mode: packages.NeedModule,
 	}
+	var crdPaths []string
 	for dep, crdDirs := range externalDeps {
-		depPath, err := exec.Command("go", "list", "-m", "-f", "{{ .Path }}@{{ .Version }}", dep).Output()
-		if err != nil {
-			return crdPaths, err
-		}
 		for _, crdDir := range crdDirs {
-			crdPaths = append(crdPaths, filepath.Join(strings.TrimSuffix(string(gopath), "\n"),
-				"pkg", "mod", strings.TrimSuffix(string(depPath), "\n"), crdDir))
+			pkgs, err := packages.Load(packageConfig, dep)
+			if err != nil {
+				return nil, err
+			}
+			pkg := pkgs[0]
+			if pkg.Errors != nil {
+				return nil, pkg.Errors[0]
+			}
+			crdPaths = append(crdPaths, filepath.Join(pkg.Module.Dir, crdDir))
```

Comment:
Seems like https://github.com/vmware-tanzu/tanzu-framework/blob/main/addons/pinniped/tanzu-auth-controller-manager/controllers/suite_test.go#L104 has a similar function ```getExternalCRDPaths()```

Reply:
yes, it will. thanks for pointing out.

Reply:
Done